### ISSUE TITLE
[FEATURE] Embed Flowser in "Interact" page

### ISSUE THREAD

**Issue by bartolomej** [muttoni/contract-browser#9]
### Issue to be solved

Currently the "Interact" page is not yet implemented. I believe it was meant to host a UI for interacting with the contract shown on the overview page.

### Suggest A Solution

We at Flowser are already working on a web-based tool for easy interactions with the contracts on Flow blockchain with neat features like FLIX integration: https://interact.flowser.dev

I believe this is a perfect tool to for integration in contract browser "Interact" page.

**Comment by bartolomej** [comment]
We actually already started working on this and have a WIP design specs and PR here: https://github.com/onflowser/contract-browser/pull/1

We also predicted this integration in our latest grant ("adoption" milestone): https://github.com/onflow/developer-grants/issues/260

**Comment by muttoni** [comment]
Could flowser expose an API with common scripts and contracts involved, and Contract Browser can submit a search for a given contract being loaded, and suggest the common interactions.

e.g.

- I search "nonfungibletoken"
- CB sends a request to Flowser with that contract/address
- flowser interact has 10 scripts / tx's that use NonFungibleToken
- flowser responds with those 10 scripts with pre-agreed on metadata
- CB lists those interactions out, with links out to Flowser
- Ideally in the future, flowser interact exposes an iframe and I can integrate in contract browser directly.

**Comment by muttoni** [comment]
In the meantime I've pointed "interact" tab to point to Flowser :)

![image](https://github.com/muttoni/contract-browser/assets/27052451/999bd0e3-29a2-4ca6-b434-f1138c2daf2c)

will include in next release (as you can tell from screenshot working on Crescendo stuff!

**Comment by muttoni** [comment]
This is now live! View it in action :)

**Comment by bartolomej** [comment]
> Could flowser expose an API with common scripts and contracts involved, and Contract Browser can submit a search for a given contract being loaded, and suggest the common interactions.
> 
> e.g.
> 
> * I search "nonfungibletoken"
> * CB sends a request to Flowser with that contract/address
> * flowser interact has 10 scripts / tx's that use NonFungibleToken
> * flowser responds with those 10 scripts with pre-agreed on metadata
> * CB lists those interactions out, with links out to Flowser
> * Ideally in the future, flowser interact exposes an iframe and I can integrate in contract browser directly.

Yes, this is definitely possible. Do you think it would be better to:
- have a mixed experience where contractbrowser fetches and displays a preview of available interactions within contractbrowser.com and then links out to flowser in a separate page
- or have the flowser directly embedded within contractbrowser.com as an iframe (as you mentioned in the last point)?

I think both would be possible with what we are trying to build tho. We are thinking the iframe embed approach would be the primary way to embed flowser in 3rd party websites, as we also proposed [in the grant](https://github.com/onflow/developer-grants/issues/260#:~:text=if%20applicable%2C%20add%20API%20to%20allow%20websites%20to%20embed%20and%20control%20the%20playground%20(e.g.%20execute%20a%20Flow%20CLI%20command%2C%20navigate%20to%20a%20different%20UI%20view%2C%20paste%20Cadence%20code%20to%20the%20editor%2C%E2%80%A6)).

**Comment by bartolomej** [comment]
> In the meantime I've pointed "interact" tab to point to Flowser :)
> ...

Wow awesome update, love it!

**Comment by muttoni** [comment]
> Yes, this is definitely possible. Do you think it would be better to:
> 
> * have a mixed experience where contractbrowser fetches and displays a preview of available interactions within contractbrowser.com and then links out to flowser in a separate page
> * or have the flowser directly embedded within contractbrowser.com as an iframe (as you mentioned in the last point)?
> 
> I think both would be possible with what we are trying to build tho. We are thinking the iframe embed approach would be the primary way to embed flowser in 3rd party websites, as we also proposed [in the grant](https://github.com/onflow/developer-grants/issues/260#:~:text=if%20applicable%2C%20add%20API%20to%20allow%20websites%20to%20embed%20and%20control%20the%20playground%20(e.g.%20execute%20a%20Flow%20CLI%20command%2C%20navigate%20to%20a%20different%20UI%20view%2C%20paste%20Cadence%20code%20to%20the%20editor%2C%E2%80%A6)).

Open to either option - I think from a UX perspective perhaps embedding via iframe is better because it avoids unnecessary back and forths between sites.
